public/app/core/services/search_srv.ts

```diff
@@ -16,8 +17,8 @@ export class SearchSrv {
   starredIsOpen: boolean;
 
   constructor() {
-    this.recentIsOpen = store.getBool('search.sections.recent', true);
-    this.starredIsOpen = store.getBool('search.sections.starred', true);
+    this.recentIsOpen = store.getBool(`${SECTION_STORAGE_KEY}.recent`, true);
+    this.starredIsOpen = store.getBool(`${SECTION_STORAGE_KEY}.starred`, true);
```

Comment:
Folder state changes don't appear until dashboard reload, because `recentIsOpen` and `starredIsOpen` initialized only on page load. I guess it should be in `getStarred()` and `getRecentDashboards()` methods respectively.

Reply:
I think the idea here is to keep the expanded/collapsed state on page reload for Starred and Recent folders. Otherwise it should behave like a normal toggle. Or did you mean something else?

Reply:
How to replicate:
1. Open search.
1. Starred folder expanded. Collapse it.
1. Close search.
1. Open it again.

In Grafana 6.7 you'll see starred folder collapsed, but in current version it's expanded  anyway.

Reply:
Ah, I see. Good catch!